client/transaction-pool/src/lib.rs

```diff
@@ -429,22 +435,95 @@ impl<N: Clone + Copy + AtLeast32Bit> RevalidationStatus<N> {
 	}
 }
 
+/// Find the blocks that were retracted while re-orging the chain
+/// from the old best to `new_best`. It will collect all block
+/// hashes that are reached by walking from all leaves to the common
+/// anchestor that are not part of the canonical chain. The given
+/// `tree_route` must be the one that was calculated on import.
+///
+/// Given the following:
+///
+/// ```ignore
+///     -> D0 (old best, tx0)
+///    /
+/// C - -> D1 (tx1)
+///    \
+///     -> D2 (new best)
+/// ```
+///
+/// This function will return `D0` and `D1`.
+fn find_retracted_blocks<'a, Block: BlockT>(
```

Comment:
Imho should be part of concrete `ChainApi` implementation - has nothing to do with transaction pool.